core/chains/evm/client/rpc_client.go

```diff
@@ -1508,9 +1393,9 @@ func (r *rpcClient) onNewFinalizedHead(ctx context.Context, requestCh <-chan str
 	}
 }
 
-func (r *rpcClient) GetInterceptedChainInfo() (latest, highestUserObservations commonclient.ChainInfo) {
-	r.chainInfoLock.RLock()
-	defer r.chainInfoLock.RUnlock()
+func (r *RPCClient) GetInterceptedChainInfo() (latest, highestUserObservations commonclient.ChainInfo) {
+	r.chainInfoLock.Lock()
+	defer r.chainInfoLock.Unlock()
```

Comment:
Why do we need a write-lock here?

Reply:
Need the lock to deal with concurrent writes to `latestChainInfo` and `highestUserObservations` since reading in Go isn't atomic.

Reply:
I understand why you need to a lock, I'm saying that I don't think it should have been changed from `RLock()` to `Lock()`.

`RLock()` will keep anything from being able to write to `latestChainInfo` or `highestUserObservations` currently.  But it's better than `Lock()` because it will will allow multiple readers at once while the writers are blocked.

Both `chainInfoLock` and `subsSliceMu` are declared as `sync.RWMutex`, but you're not using the `RLock` feature for either of those presently, so at least for `subsSliceMu` I think it makes more sense to replace it with a `sync.Mutex`.  In the case of `chainInfoLock`, we should be taking advantage of  the `RLock` feature since it would be useful here. But if we're not going to use it then we may as well reduce that one to a `sync.Mutex` also.